### PR TITLE
fix(FOROME-834): ability to change approx in compound_req, comp_het

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-834

### CODE REVIEW

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts [hunk 1]
```diff
@@ -18,69 +27,79 @@ import { getSortedArray } from '@utils/getSortedArray'
...
   constructor() {
     makeAutoObservable(this)
   }
 
   // root functions
 
-  public setCurrentMode(modeType: ModeTypes) {
-    this.currentMode = modeType
+  public get requestCondition(): TRequestCondition[] {
+    return this._requestCondition
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts]: just make requestCondition public

File: src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts [hunk 2]
```diff
@@ -29,17 +36,12 @@ class ModalCompoundHetStore {
     this.currentMode = undefined
   }
 
-  public setStateCondition(condtition: string): void {
-    this.stateCondition = condtition
+  public get approx(): ApproxNameTypes {
+    return this._approx
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts]: just make approx public

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts [hunk 3]
```diff
@@ -18,69 +27,79 @@ import { getSortedArray } from '@utils/getSortedArray'
...
+  }
+
+  public get activeRequestIndex(): number {
+    return this._activeRequestIndex
   }
 
-  public setRequestCondition(requestCondition: ({}[] | [number, any])[]): void {
-    this.requestCondition = requestCondition
+  public get currentMode(): ModeTypes | undefined {
+    return this._currentMode
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts]: same

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts [hunk 4]
```diff
@@ -18,69 +27,79 @@ import { getSortedArray } from '@utils/getSortedArray'
...
   }
 
-  public setStateCondition(stateCondition: string): void {
-    this.stateCondition = stateCondition
+  public get approx(): ApproxNameTypes {
+    return this._approx
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts]: same

File: src/pages/filter/ui/panels/function-panel/components/compound-request/compound-request.store.ts [hunk 5]
```diff
@@ -23,6 +26,9 @@ class CompoundRequestStore {
   private _resetValue: string = ''
   private _activeRequestIndex = this.requestCondition.length - 1
   private _currentMode?: ModeTypes
+  private _approx = datasetStore.isXL
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/compound-request/compound-request.store.ts]: Looks like somebody switched from C# :) This _privateProp thing doesn't make sense in TS. If you create new props, avoid it.

**Review by QYaroslavTrefilov — APPROVED** [review]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**